### ISSUE TITLE
Add support for end of line comments to function-comma-newline-after

### ISSUE THREAD

**Issue by Nirazul** [stylelint/stylelint#1022]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

I've got a scss list with comments at the end of the line. As with `block-closing-brace-newline-after`, this is not allowed by the rule. There are two things that struck me:
- The rule is triggered by the comment
- The scss list is handled as a function (although I'm not sure if that's intended behaviour)

I've prepared failing rules for those cases, but wanted to check first if they are regarded as errors.

> Which rule, if any, is this issue related to?

`function-comma-newline-after`

> What CSS is needed to reproduce this issue?

e.g.

```css
  $colors: (
    red, /* stylelint-disable-line color-named */
    #fff,
    #ddd,
  );
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "function-comma-newline-after": "always-multi-line"
  }
}
```

> Which version of stylelint are you using?

`4.3.2`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

gulp-stylelint

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Related to SCSS lists

> What did you expect to happen?

"No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

"The following warnings were flagged:

```shell
error  Expected newline after "," in a multi-line function   function-comma-newline-after
```

**Comment by davidtheclark** [comment]
I'm ok with saying that `function-comma-newline-after` should ignore things in parentheses that don't have a word immediately preceding them — such as a Sass map. So if we do that, your specific problem will go away.

But the general problem remains, in case somebody did the same in a real function. So I'd agree that that's a thing we want to fix. Generally, where we're expecting a newline, we should also accept a space followed by a single-line comment followed by a newline.

Any disagreement?

**Comment by davidtheclark** [comment]
I just pushed a commit that should make `function-comma-newline-after` ignore Sass lists. So now it's just left to get the comments handled properly with real functions, I think.

**Comment by jeddy3** [comment]
> Any disagreement?

Nope, sgtm.

For anyone wanting to help with this enhancement, you'll find some pointers in [`selector-list-comma-newline-after`](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-list-comma-newline-after/index.js#L31). It allows end of line comments after the comma in a selector list.

**Comment by Nirazul** [comment]
fyi I've started working on this. 
I'm still struggling to set breakpoints in the code, so "working" is a bit far fetched ^^
